packages/pwa-kit-dev/src/configs/webpack/plugins/overrides-resolver-plugin.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import resolve from 'resolve/sync'
+import {buildCandidatePaths} from '../../../utils/resolver-utils'
+
+export const DEFAULT_FILE_EXTENSIONS = ['.ts', '.tsx', '.js', '.jsx', '.json']
+
+/**
+ * @class OverridesResolverPlugin
+ *
+ *  This plugin provides the "overrides" behavior of the application extensibility feature. This allows
+ *  App Extension developers to define which files are part of their public api thus can be overridden, but
+ *  also the module resolution algorithm.
+ */
+class OverridesResolverPlugin {
+    constructor(options) {
+        this.projectDir = options.projectDir?.replace(/\\/g, '/')
```

Comment:
I understand that we are replacing a potential Windows-style path, that uses backslashes, with a Unix-style path that uses forward slashes.

Did we test this regular expression on a Windows VM? Should we use a more native Node method like `path` instead? E.g. `path.posix.normalize` https://nodejs.org/api/path.html#pathnormalizepath.

Reply:
What happens when `options.projectDir` is empty? Should we default to `process.cwd()`?